src/main/java/duke/utility/Parser.java

```diff
@@ -0,0 +1,164 @@
+package duke.utility;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.UnmarkCommand;
+import duke.exceptions.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * Parses inputs to Duke.
+ */
+public class Parser {
+    private static Event createEvent(String s) throws DukeException {
+        int fromIndex = s.indexOf("/from");
+        int toIndex = s.indexOf("/to");
+        if (fromIndex == -1 || toIndex == -1 || s.length() < 7) {
+            throw new DukeException("Format Error, Event must be in format:"
+                    + "Event /from YYYY-MM-DD HHmm /to YYYY-MM-DD HHmm");
+        }
+        String eventName = s.substring(6, fromIndex - 1);
+        try {
+            String from = s.substring(fromIndex + 6, toIndex - 1);
+            String to = s.substring(toIndex + 4);
+            return new Event(eventName, from, to);
+        } catch (StringIndexOutOfBoundsException | DateTimeParseException e) {
+            throw new DukeException("Format Error, Event must be in format:"
+                    + "Event /from YYYY-MM-DD HHmm /to YYYY-MM-DD HHmm");
+        }
+    }
+
+    private static Deadline createDeadline(String s) throws DukeException {
+        int byIndex = s.indexOf("/by");
+        if (byIndex == -1 || s.length() < 10) {
+            throw new DukeException("Format Error, Deadline must be in format: Deadline /by YYYY-MM-DD HHmm");
+        }
+        String deadlineName = s.substring(9, byIndex - 1);
+        try {
+            String deadlineBy = s.substring(byIndex + 4);
+            return new Deadline(deadlineName, deadlineBy);
+        } catch (StringIndexOutOfBoundsException | DateTimeParseException e) {
+            throw new DukeException("Format Error, Deadline must be in format: Deadline /by YYYY-MM-DD HHmm");
+        }
+    }
+
+    private static Todo createTodo(String s) throws DukeException {
+        if (s.length() < 6) {
+            throw new DukeException("Error, description of todo is missing");
+        }
+        String name = s.substring(5);
+        return new Todo(name);
+    }
```

Comment:
createEvent, createDeadline, createTodo seem to have some duplicated code. Maybe can be combined into a single method for simplicity?